azkaban-common/src/main/java/azkaban/executor/ExecutableFlow.java

```diff
@@ -86,7 +86,11 @@ public class ExecutableFlow extends ExecutableFlowBase {
   private String executionPath;
   private ExecutionOptions executionOptions;
   private double azkabanFlowVersion;
+  // flow lock, once locked, the flow won't be able to execute/be modified/be scheduled...
   private boolean isLocked;
+  // If the flow is a part of production project, i.e. the project is enabled with UploadLock,
+  // then the generated executable flow will mark as production flow.
+  private boolean isProductionFlow;
```

Comment:
perhaps a better name for this is something along the lines of isProjectLockable?